File: states.js

var IDLE = {
  id: 0,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_IDLE = 0;
var OFF = {
  id: 1,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_OFF = 1;
var ROBOTICARM = {
  id: 2,
  master: {
    values: {
      
    },
    events: {
      moveToHigh: function moveToHigh() { manager.sendEvent(0, 2); },
      moveToLow: function moveToLow() { manager.sendEvent(1, 2); },
      magnetOn: function magnetOn() { manager.sendEvent(2, 2); },
      magnetOff: function magnetOff() { manager.sendEvent(3, 2); },
      armUp: function armUp() { manager.sendEvent(4, 2); },
      armDown: function armDown() { manager.sendEvent(5, 2); },
      goHome: function goHome() { manager.sendEvent(6, 2); }
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_ROBOTICARM = 2;

var STATES = {
  IDLE: IDLE,
  OFF: OFF,
  ROBOTICARM: ROBOTICARM
};
var manager = new Manager([IDLE, OFF, ROBOTICARM]);

File: RoboticArmAMIB1/states.cpp
#include "states.h"

static const StateInfo state_infos[3] = {
  {IDLE::setup, IDLE::enter, IDLE::exit, IDLE::loop, IDLE::event},
  {OFF::setup, OFF::enter, OFF::exit, OFF::loop, OFF::event},
  {ROBOTICARM::setup, ROBOTICARM::enter, ROBOTICARM::exit, ROBOTICARM::loop, ROBOTICARM::event}
};

static const WireValue wire_values[0] = {
  
};

MasterManager<State, 3, 0> manager(0xb1928129, state_infos, wire_values, 0);

namespace IDLE {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace OFF {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace ROBOTICARM {


void event(uint8_t ev) {
  switch (ev) {
  case 0:
    events::moveToHigh();
    break;
  case 1:
    events::moveToLow();
    break;
  case 2:
    events::magnetOn();
    break;
  case 3:
    events::magnetOff();
    break;
  case 4:
    events::armUp();
    break;
  case 5:
    events::armDown();
    break;
  case 6:
    events::goHome();
    break;
  default:
    break;
  }
}


}

